bin/templates/cordova/lib/build.js

```diff
@@ -422,11 +477,15 @@ function parseOpts(options, resolvedTarget) {
 module.exports.runClean = function(options) {
     var opts = parseOpts(options);
     var builder = builders[opts.buildMethod];
-    return builder.prepEnv()
+    return builder.prepEnv(opts)
     .then(function() {
-        return builder.clean(opts.extraArgs);
+        return builder.clean(opts);
     }).then(function() {
         shell.rm('-rf', path.join(ROOT, 'out'));
+    }).then(function() {
+        ['debug', 'release'].forEach(function(config) {
+            removeIfExists(path.join(ROOT, config + SIGNING_PROPERTIES));
```

Comment:
I was really hoping to get the scenario of CLI + IDE (Android studio) interop working. If they both consume the same .properties file they can interoperate. The user could create a build.json file with the correct values and would be able to achieve this. I'm getting inclined to delete the file _only_ when it is autogenerated.